### PR TITLE
fix: update/clean/format the miner code, rewrite the docs, etc

### PR DESCRIPTION
**Pull request by agiscientist**
Hi @yang-han,

There are several updates and improvements in this PR:

- **Refactored `miner.py`**: Cleaned and streamlined the code for better readability and maintainability.
- **Documentation updates**: Updated related documentation to reflect the recent changes.
- **Code formatting**: Formatted the `validator.py` and `miner.py` for more clear usages.

Let me know if there’s anything else you’d like me to adjust.

### CODE REVIEW

File: neurons/miner.py [hunk 1]
```diff
@@ -43,158 +43,38 @@
...
-        query.results = ranked_docs
-        end_time = datetime.now()
-        elapsed_time = (end_time - start_time).total_seconds()
-        bt.logging.info(
-            f"processed SemanticSearchSynapse in {elapsed_time} seconds",
-        )
-        return query
+        pass
 
     async def forward_discord_search(
```
Comment (webroot-kaito) [inline neurons/miner.py:67]: nit: let's add some comments to those tasks deprecated

File: quickstart.md [hunk 2]
```diff
@@ -406,3 +281,136 @@ To monitor your validator process, use the following pm2 commands to monitor the
 pm2 status
 pm2 logs <id>
 ```
+
+
+## Other Setup (Reference Only in This Version):
```
Comment (webroot-kaito) [inline quickstart.md:286]: Maybe add a few comments indicating "this task is no longer being distributed in the current version. enable this setup only if you need to support vector retrieval tasks"

**Review by webroot-kaito — APPROVED** [review]

File: neurons/miner.py (hunk 1 above)
Comment (agiscientist) [reply neurons/miner.py:71]: solved by update

File: quickstart.md (hunk 2 above)
Comment (agiscientist) [reply quickstart.md]: solved by a new update

**Pull request merged**